ksql-execution/src/test/java/io/confluent/ksql/execution/util/ExpressionTypeManagerTest.java

```diff
@@ -0,0 +1,490 @@
+import static io.confluent.ksql.execution.testutil.TestExpressions.COL3;
+import static io.confluent.ksql.execution.testutil.TestExpressions.COL7;
+import static io.confluent.ksql.execution.testutil.TestExpressions.literal;
+import static io.confluent.ksql.execution.testutil.TestExpressions.MAPCOL;
+import static io.confluent.ksql.execution.testutil.TestExpressions.SCHEMA;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.ArgumentMatchers.anyList;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.ImmutableList;
+import io.confluent.ksql.execution.expression.tree.ArithmeticBinaryExpression;
+import io.confluent.ksql.execution.expression.tree.BooleanLiteral;
+import io.confluent.ksql.execution.expression.tree.ComparisonExpression;
+import io.confluent.ksql.execution.expression.tree.ComparisonExpression.Type;
+import io.confluent.ksql.execution.expression.tree.Expression;
+import io.confluent.ksql.execution.expression.tree.FunctionCall;
+import io.confluent.ksql.execution.expression.tree.InListExpression;
+import io.confluent.ksql.execution.expression.tree.InPredicate;
+import io.confluent.ksql.execution.expression.tree.IntegerLiteral;
+import io.confluent.ksql.execution.expression.tree.LikePredicate;
+import io.confluent.ksql.execution.expression.tree.NotExpression;
+import io.confluent.ksql.execution.expression.tree.QualifiedName;
+import io.confluent.ksql.execution.expression.tree.SearchedCaseExpression;
+import io.confluent.ksql.execution.expression.tree.SimpleCaseExpression;
+import io.confluent.ksql.execution.expression.tree.StringLiteral;
+import io.confluent.ksql.execution.expression.tree.SubscriptExpression;
+import io.confluent.ksql.execution.expression.tree.TimeLiteral;
+import io.confluent.ksql.execution.expression.tree.TimestampLiteral;
+import io.confluent.ksql.execution.expression.tree.WhenClause;
+import io.confluent.ksql.function.FunctionRegistry;
+import io.confluent.ksql.function.KsqlFunction;
+import io.confluent.ksql.function.UdfFactory;
+import io.confluent.ksql.function.udf.structfieldextractor.FetchFieldFromStruct;
+import io.confluent.ksql.schema.Operator;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import io.confluent.ksql.schema.ksql.types.SqlType;
+import io.confluent.ksql.schema.ksql.types.SqlTypes;
+import io.confluent.ksql.util.KsqlException;
+import java.util.Optional;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+@SuppressWarnings("OptionalGetWithoutIsPresent")
+public class ExpressionTypeManagerTest {
```

Comment:
Unfortunately this is showing up as a whole new test. It's not, I've just rewritten the cases to build their own Expression instances instead of having the engine parse them.